### DISCUSSION TITLE
Make Renovate work on many folders inside monorepo

### DISCUSSION THREAD

**Discussion by 0xmkh** [renovatebot/renovate#20117] [Request Help]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, tell us what version of Renovate you run.

not self-hosted

### If you're self-hosting Renovate, select which platform you are using.

None

### If you're self-hosting Renovate, tell us what version of the platform you run.

not self hosted

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Wanted end result.

My project looks like this.

I have 4 monorepos.

project:
 - /frontend
 - /backend
 - /helpers
 - /utils
 
I want to make sure the Renovate only upgrades specific packages for specific folders.

How do i make this work ? On `/frontend` i only want Renovate to upgrade `axios` and nothing else. On `backend` i want Renovate to only upgrade `typescript` and that's it. Nothing else should be upgraded on other 2 folders.

Any code or suggestions ?

### What you tried so far.

Tried `matchPackageNames` and `matchPaths`

**Comment by rarkins** [comment]
Please provide the full config you tried

**Comment by 0xmkh** [comment]
Project looks like this:
```
- package.json
  /frontend
  - package.json
  /backend
  - package.json
```

Here: 
```
{
  "extends": ["config:base"],
  "enabledManagers": ["npm"],
  "packageRules": [
    {
      "matchPaths": ["/package.json"],
      "matchPackageNames": ["typescript"],
      "automerge": false
    },
    {
      "matchPaths": ["frontend/**"],
      "matchPackageNames": ["axios"],
      "automerge": false
    },
    {
      "matchPaths": ["backend/**"],
      "matchPackageNames": ["jest"],
      "automerge": false
    }
  ]
}
```

Renovate.json is located in the root

**Reply by rarkins** [reply]
The only thing the above config is doing is disabling automerge for selective packages. It doesn't seem to match at all with what you described earlier (which seemed about raising PRs only for selective packages). Also.. automerge is already disabled by default so the packageRules literally achieve nothing by disabling something which is already disabled.

**Reply by 0xmkh** [reply]
Ok i will disable automerge. Can you tell me how i can select spesific packages to update in a monorepo ? `matchPackageNames` is the only configuration i've found in the documentation. A example code would be appreciated 👍

**Reply by rarkins** [reply]
Let's say those packages are the ones you want ENABLED and everything else disabled. It would look like this:

```
{
  "extends": ["config:base"],
  "enabledManagers": ["npm"],
  "packageRules": [
    {
      "matchPackagePatterns": ["*"],
      "enabled": false
    },
    {
      "matchPaths": ["/package.json"],
      "matchPackageNames": ["typescript"],
      "enabled": true
    },
    {
      "matchPaths": ["frontend/**"],
      "matchPackageNames": ["axios"],
      "enabled": true
    },
    {
      "matchPaths": ["backend/**"],
      "matchPackageNames": ["jest"],
      "enabled": true
    }
  ]
}
```

i.e. disable every package, then selectively re-enable the ones you want.

**Reply by 0xmkh** [reply]
Oh nice! That's perfect will try doing that. Is enabled `false` needed ? It's `false` by default right ? I guess that's all i needed actually really appreciate the help 👍

**Reply by rarkins** [reply]
It's enabled by default. By default Renovate updates all dependencies it can!
